File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
import java.util.*;

class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        String s = Integer.toString(x);
        List<Integer> numList = new ArrayList<>();
        for(int i = 0; i < s.length(); i++){
            numList.add(s.charAt(i) - '0');
        }
        int num = 0;
        for(int i = 0; i < numList.size(); i++){
            num += numList.get(i);
        }
        if(x % num != 0){
            answer = false;
        }
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.swift
func solution(_ x:Int) -> Bool {
    var result = true
    
    var x_str = String(x)
    
    var sum = 0
    
    for i in x_str {
        if let num = Int(String(i)){
            sum += num
        }
    }
    
    if (x % sum != 0) {
        result = false
    }
    
    return result
}